pkg/scheduler/schedule_one.go

```diff
@@ -106,14 +130,57 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		metrics.Goroutines.WithLabelValues(metrics.Binding).Inc()
 		defer metrics.Goroutines.WithLabelValues(metrics.Binding).Dec()
 
-		sched.bindingCycle(bindingCycleCtx, state, fwk, scheduleResult, assumedPodInfo, podsToActivate, start)
+		status := sched.bindingCycle(bindingCycleCtx, state, fwk, scheduleResult, assumedPodInfo, start)
+		if !status.IsSuccess() {
+			var reason string
+			if status.IsUnschedulable() {
+				metrics.PodUnschedulable(fwk.ProfileName(), metrics.SinceInSeconds(start))
+				reason = v1.PodReasonUnschedulable
+			} else {
+				metrics.PodScheduleError(fwk.ProfileName(), metrics.SinceInSeconds(start))
+				reason = v1.PodReasonSchedulerError
+			}
+
+			assumedPod := assumedPodInfo.Pod
+			// trigger un-reserve plugins to clean up state associated with the reserved Pod
+			fwk.RunReservePluginsUnreserve(ctx, state, assumedPod, scheduleResult.SuggestedHost)
+			if forgetErr := sched.Cache.ForgetPod(assumedPod); forgetErr != nil {
+				klog.ErrorS(forgetErr, "scheduler cache ForgetPod failed")
+			} else {
+				// "Forget"ing an assumed Pod in binding cycle should be treated as a PodDelete event,
+				// as the assumed Pod had occupied a certain amount of resources in scheduler cache.
+				//
+				// Avoid moving the assumed Pod itself as it's always Unschedulable.
+				// It's intentional to "defer" this operation; otherwise MoveAllToActiveOrBackoffQueue() would
+				// update `q.moveRequest` and thus move the assumed pod to backoffQ anyways.
+				if status.IsUnschedulable() {
+					defer sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, func(pod *v1.Pod) bool {
+						return assumedPod.UID != pod.UID
+					})
+				} else {
+					sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, nil)
+				}
+			}
+
+			sched.FailureHandler(ctx, fwk, assumedPodInfo, status.AsError(), reason, clearNominatedNode)
+			return
+		}
+
+		// At the end of a successful binding cycle, move up Pods if needed.
+		if len(podsToActivate.Map) != 0 {
```

Comment:
Moved this out of schedulingCycle, `podsToActivate` is a value of pointer, I think it's ok.